pkg/odo/genericclioptions/context.go

```diff
@@ -147,6 +169,46 @@ func newContext(command *cobra.Command, createAppIfNeeded bool) *Context {
 			}
 		}
 	}
+	return app
+}
+
+// resolveComponent resolves component
+func resolveComponent(command *cobra.Command, configFileName string, context *Context) string {
+	var cmp string
+	lci, err := config.NewLocalConfigInfo(configFileName, false)
+	util.LogErrorAndExit(err, "could not get component settings from config file")
```

Comment:
I would like all the logging to be doing by the `GenericRun` and these function can just return error. just a suggestion. maybe @kadel can confirm this

Reply:
Agreed. But that will require a bigger refactor as this is pattern in most of the file. Maybe an issue to track it